easybuild/easyblocks/e/esmf.py

```diff
@@ -126,6 +127,30 @@ def configure_step(self):
                         netcdf_libs.append('-lnetcdf_c++')
                 env.setvar('ESMF_NETCDF_LIBS', ' '.join(netcdf_libs))
 
+        if self.cfg['external_pio']:
+            if LooseVersion(self.version) >= LooseVersion('8.4.0'):
+                pio = get_software_root('ParallelIO')
+                if pio:
+                    if LooseVersion(get_software_version('ParallelIO')) >= LooseVersion('2.5.9'):
+                        env.setvar('ESMF_PIO', 'external')
```

Comment:
There is no real need for an explicit 'external_pio' option here, just change the code to something like
```
pio = get_software_root('ParallelIO')
if pio:
  if LooseVersion(self.version) >= LooseVersion('8.4.0'):
    check pio version etc
  else:
     msg = "External PIO is not supported for EMSF versions < 8.4.0"
    ...
```

Reply:
@akesandgren Oh, ok. I just wanted this to be a bit more explicit than specifying one dependency in the config file. external LAPACK can be done the same way.
Also, quick question: do I understand that `env.setvar()` is passing these vars as build_opts for CMake?

Reply:
LAPACK is different since it is part of the toolchain (foss/intel) and will always be there, hence there needs to be a way to disable it.

env.setenv just sets an environment variable.
The EMSF easyblock does not (currently) use the CMake building of ESMF, just the plain Makefile

Reply:
Removed the option and check for dependency instead.